### PR TITLE
fix(toaster): let on-demand toasts know the toaster has been destroyed

### PR DESCRIPTION
**Pull request by tmorehouse**
### Describe the PR

Prevent toasts from re-appearing when a toaster has been destroyed and a new one with the same
name has been created.  Also prevents possible memory leaks when toasters are destroyed.

Also includes updates to the toaster SCSS for better support of smaller screens

### PR checklist

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Enhancement
- [ ] ARIA accessibility
- [ ] Documentation update
- [ ] Other (please describe)

**Does this PR introduce a breaking change?** (check one)

- [x] No
- [ ] Yes (please describe)

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch, **not** the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (i.e. `[...] (fixes #xxx[,#xxx])`, where "xxx" is the issue number)
- [x] It should address only one issue or feature. If adding multiple features or fixing a bug and adding a new feature, break them into separate PRs if at all possible.
- [x] The title should follow the [**Conventional Commits**](https://www.conventionalcommits.org/) naming convention (i.e. `fix(alert): not alerting during SSR render`, `docs(badge): update pill examples, fix typos`, `chore: fix typo in README`, etc). **This is very important, as the `CHANGELOG` is generated from these messages.**

**If new features/enhancement/fixes are added or changed:**

- [ ] Includes documentation updates (including updating the component's `package.json` for slot and event changes)
- [ ] New/updated tests are included and passing (if required)
- [x] Existing test suites are passing
- [x] The changes have not impacted the functionality of other components or directives
- [ ] ARIA Accessibility has been taken into consideration (Does it affect screen reader users or keyboard only users? Clickable items should be in the tab index, etc.)

**If adding a new feature, or changing the functionality of an existing feature, the PR's
description above includes:**

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

### CODE REVIEW

File: src/components/toast/_toaster.scss [hunk 1]
```diff
@@ -1,25 +1,31 @@
...
+  max-width: $b-toaster-max-width;
   overflow: visible;
   z-index: $b-toaster-zindex;
 
   // Ensure that an empty toaster isn't occupying any space
   // and obscuring access to underlying elements
   &:empty {
     padding: 0 0;
+    margin: 0 0;
+    display: none;
```
Comment (jackmu95) [inline src/components/toast/_toaster.scss]: Why reset the `padding` and `margin` when we set it to `display: none;` anyway?

Comment (tmorehouse) [reply src/components/toast/_toaster.scss]: Just in case users have defined their own toasters definition.

Comment (tmorehouse) [reply src/components/toast/_toaster.scss]: Was also going to add in `width: 0 ; height: 0;`

Comment (tmorehouse) [reply src/components/toast/_toaster.scss]: The margin and padding removal is to make sure the toaster container is moved to the very edge of the screen.

**Comment by tmorehouse** [comment]
After playing a bit with the styles and positioning, I'm starting to lean towards adding an extra wrapper div inside the `b-toaster` to allow for more positioning options.

**Pull request merged**